### PR TITLE
server: expose randomgenerator via the api interface

### PR DESCRIPTION
**Pull request by akonradi**
Commit Message: Expose RandomGenerator via the Api interface.
Additional Description:
Api is a natural fit since randomness is often exposed from the platform. This PR
is the first step towards #13243, and only adds an additional access point for the
RandomGenerator in Api; later PRs will clean up other accesses via
Server::random().

Risk Level: low
Testing: built and ran affected tests
Docs Changes: none
Release Notes: n/a

### CODE REVIEW

File: test/test_common/utility.cc [hunk 1]
```diff
@@ -431,8 +439,8 @@ ApiPtr createApiForTest(Event::TimeSystem& time_system) {
 }
 
 ApiPtr createApiForTest(Stats::Store& stat_store, Event::TimeSystem& time_system) {
-  return std::make_unique<Impl>(Thread::threadFactoryForTest(), stat_store, time_system,
-                                Filesystem::fileSystemForTest());
+  return std::make_unique<TestImpl>(Thread::threadFactoryForTest(), stat_store, time_system,
```
Comment (asraa) [inline test/test_common/utility.cc:442]: huh, nice catch!

**Review by asraa — APPROVED** [review]
Nice, LGTM! Thanks for working on this.

@envoyproxy/senior-maintainers can you please merge?

**Comment by mattklein123** [comment]
Coverage issue is legit, PTAL.

/wait

**Comment by akonradi** [comment]
Fixed coverage issue that was caused by the method I added never being called 🤦

**Review by mattklein123 — DISMISSED** [review]
Thanks!

**Comment by mattklein123** [comment]
This is also failing on the example_config_test. cc @phlax @ggreenway @htuch 

We will need to figure out why this is breaking here and not in main.

/wait

**Comment by phlax** [comment]
@mattklein123 i think the failure here is to do with changes in the PR - heres the azure output:

```
TestRandomGenerator running with seed 745623360
unknown file: Failure
C++ exception with description "Function call: randomGenerator
    The mock function has no default action set, and its return type has no default value set." thrown in the test body.
[  FAILED  ] ExampleConfigsTest.All (123 ms)

```

**Comment by mattklein123** [comment]
Ah OK, apologies. False alarm! Then unclear on @ggreenway issue.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**